Source/FakeItEasy/Core/ArgumentValueFormatter.cs

```diff
@@ -31,11 +32,7 @@ public virtual string GetArgumentValueAsString(object argumentValue)
             var argumentType = argumentValue.GetType();
 
             IArgumentValueFormatter formatter;
-            if (!this.cachedFormatters.TryGetValue(argumentType, out formatter))
-            {
-                formatter = this.ResolveTypeFormatter(argumentType);
-                this.cachedFormatters.Add(argumentType, formatter);
-            }
+            formatter = this.cachedFormatters.GetOrAdd(argumentType, ResolveTypeFormatter);

```

Comment:
I'm away from a compiler now, but I think `ResultTypeFormatter` is a style warning (no `this.`). 
If you come back in to remove the new test, how do you feel about cleaning it up while you're here?
Alternatively, I can do later.